global_fusion/src/gnss_preprocessor/gnss_preprocessor.cpp

```diff
@@ -105,7 +105,12 @@ int main(int argc, char **argv)
 	prcopt.posopt[4] = 0;               // use RAIM FDE (qmo)  1
 	prcopt.tropopt = TROPOPT_SAAS;        // troposphere option: Saastamoinen model
 	prcopt.ionoopt = IONOOPT_BRDC;		// ionosphere option: Broad cast
-	prcopt.sateph = EPHOPT_PREC;			// ephemeris option: broadcast ephemeris
+	if (std::find(satellites.begin(), satellites.end(), "SP3") != satellites.end() or std::find(satellites.begin(), satellites.end(), "SBAS") != satellites.end()){

```

Comment:
In the case of SBAS,  "brdc+sbas" should be selected.  The option "precise" is only for SP3.